### PR TITLE
[DO NOT MERGE YET] JBPM-6627 - Extend tests for Kie Server Controller Integration tests

### PR DESCRIPTION
**Pull request by jakubschwan**
Tests for JBPM-6243
@cristianonicolai Can you take a look?

### CODE REVIEW

**Comment by jakubschwan** [comment]
Jenkins retest this

**Comment by cristianonicolai** [comment]
@jakubschwan overall looks good to me. I was only wondering if you could get a green build by disabling those tests that are known to not be supported in the REST API yet. That way we could enable it as we include the new endpoints.

**Comment by jakubschwan** [comment]
Sure, in REST API are now not supported REST endpoints for Rule capability and Runtime monitoring, right? But there still be some tests failures because of web socket test (e.g. not thrown exception on wrong method entrance - not existing container)
I'll push changes later today.

**Comment by jakubschwan** [comment]
Test failures:
- RestKieControllerManagementIntegrationTest - missing REST endpoints for the SpecManagementService, implementation should be add in to the org.kie.server.controller.rest.RestSpecManagementServiceImpl
- WebSocketKieControllerManagementIntegrationTest - failed copy of template configuration - what should be copied into the new template?
- WebSocketKieControllerRuleCapabilitiesIntegrationTest - failing test with not correct entries - entry should be checked and if is not correct then exception should be thrown. + scanning not working
- WebSocketKieControllerRuntimeManagementIntegrationTest - wrong data returned for getContainers + failing test with not correct entries 

@cristianonicolai Can you check tests results too? If you think that some scenario is not right please let me know, otherwise I'll start to creating issues.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerRuntimeManagementIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    @Test
+    public void testGetServerInstances() {
+        // Create kie server template connection in controller.
+        ServerTemplate serverTemplate = createServerTemplate();
+
+        ServerInstanceKeyList serverInstances = controllerClient.getServerInstances(serverTemplate.getId());
+        assertThat(serverInstances.getServerInstanceKeys()).hasSize(1);
+        ServerInstanceKey serverInstance = serverInstances.getServerInstanceKeys()[0];
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(serverInstance.getServerName()).isEqualTo(kieServerInfo.getName());
```
Comment (jakubschwan) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerRuntimeManagementIntegrationTest.java]: @cristianonicolai What is expected name of the server instance? I think it should be same as name of the Kie Server or at least server instance should contain it. But in this test failed with message: expected:<"[KieServer@/kie-server-services]"> but was:<"[a0e625bf-251d-3c1f-ad11-0a17f1c14e21@localhost:37668]">

**Comment by mswiderski** [comment]
any plans to get this merged? if not please close

**Comment by jakubschwan** [comment]
I think it can be closed. Tests were include in PRs with fix, right @cristianonicolai ?

**Comment by cristianonicolai** [comment]
@jakubschwan correct, thanks a lot for this reproducers.

**Pull request closed without merging**